### DISCUSSION TITLE
custom cache-control in getServerSideProps not setting entire cache-control string

### DISCUSSION THREAD

**Discussion by marcusmolchany** [vercel/next.js#34938] [Help]
hey, i'm trying to add a custom `cache-control` header in `getServerSideProps`. i've followed the [next.js caching docs](https://nextjs.org/docs/going-to-production#caching) and have the following in my page component:

```ts
export async function getServerSideProps({ req, res }) {
  res.setHeader(
    'Cache-Control',
    'public, s-maxage=15, stale-while-revalidate=59'
  )

// rest of component

  return {
    props: { /* my props */ },
  }
}
```

when i go to my page in production, i see `cache-control: public` but do not see the `s-maxage=15, stale-while-revalidate=59` part of the cache-control string. is this expected?

|screenshot from postman|
|--|
|<img width="1028" alt="Screen Shot 2022-03-01 at 12 35 45 PM" src="https://user-images.githubusercontent.com/2160046/156244874-7aa5ad9b-0310-471f-9b4e-d114c5cfdceb.png">|

related issue: https://github.com/vercel/next.js/issues/33204#issuecomment-1010840976

**Comment by jstorm31** [comment]
I can relate that this happens on my site too. They may strip the other part of `cache-control` header as it is only related to CDN? I don't know so clarification would be appreciated 🙂
